projects/igniteui-angular/src/lib/grids/cell.component.ts

```diff
@@ -925,6 +925,10 @@ export class IgxGridCellComponent implements OnInit, OnChanges, OnDestroy {
         const node = this.selectionNode;
         const shouldEmitSelection = !this.selectionService.isActiveNode(node);
 
+        if (this.cellSelectionMode === GridSelectionMode.none || this.cellSelectionMode === GridSelectionMode.single) {
+            this.selectionService.clear();
+        }
+
```

Comment:
This check should not be placed in the method /as you can see in the comment there are still issues with this fix/.. try to remove it and the fix should be better performed in focusTbody() method in the grid navigation service.